File: Leetcode/Permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        self.ans = set()
        self.helper(nums, [])
        return self.ans

    def helper(self, inp, out):
        if len(inp) == 0:
            self.ans.add(tuple(out))
            return

        for i in range(len(inp)):
            ros = inp[0:i] + inp[i + 1:]
            self.helper(ros, out+[inp[i]])

File: Algo++/Recursion/Permutations.cpp
#include<iostream>
#include<cstring>
using namespace std;

void permute(string inp, string out) {
    if(inp.length() == 0) {
        cout << out << endl;
        return ;
    }

    for (int i = 0; i < inp.length(); ++i) {
        string ros = inp.substr(0,i) + inp.substr(i+1);
        permute(ros, out + inp[i]);
    }
}

int main() {
    string inp;
    cin >> inp;

    permute(inp,"");

    return 0;
}

